core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/arrayfunction/ArrayStarExpression.kt

```diff
@@ -0,0 +1,37 @@
+package ch.ergon.dope.resolvable.expression.unaliased.type.function.arrayfunction
+
+import ch.ergon.dope.DopeQuery
+import ch.ergon.dope.DopeQueryManager
+import ch.ergon.dope.resolvable.clause.ISelectOffsetClause
+import ch.ergon.dope.resolvable.expression.TypeExpression
+import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.resolvable.operator.FunctionOperator
+import ch.ergon.dope.validtype.ArrayType
+import ch.ergon.dope.validtype.ObjectType
+
+class ArrayStarExpression(
+    private val objectArray: TypeExpression<ArrayType<ObjectType>>,
+) : TypeExpression<ObjectType>, FunctionOperator {
+    override fun toDopeQuery(manager: DopeQueryManager): DopeQuery {
+        val objectArrayDopeQuery = objectArray.toDopeQuery(manager)
+        return DopeQuery(
+            queryString = toFunctionQueryString("ARRAY_STAR", objectArrayDopeQuery.queryString),
+            parameters = objectArrayDopeQuery.parameters,
+        )
+    }
+}
+
+fun arrayStar(objectArray: TypeExpression<ArrayType<ObjectType>>) = ArrayStarExpression(objectArray)
+
+@JvmName("receiverArrayStar")
+fun TypeExpression<ArrayType<ObjectType>>.arrayStar() = arrayStar(this)
```

Comment:
in the other array-functions we don't have extension-functions on `TypeExpression<ArrayType<ObjectType>>` - why do we do it with array-star?